lib/dao.js

```diff
@@ -2407,6 +2586,183 @@ DataAccessObject.prototype.unsetAttribute = function unsetAttribute(name, nullif
 };
 
 /**
+ * Replace set of attributes.
+ * Performs validation before replacing.
+ *
+ * @trigger `validation`, `save` and `update` hooks
+ * @param {Object} data Data to replace
+ * @param {Object} [options] Options for replace
+ * @param {Function} cb Callback function called with (err, instance)
+ */
+DataAccessObject.prototype.replaceAttributes = function(data, options, cb) {
+  var Model = this.constructor;
+  var id = getIdValue(this.constructor, this);
+  return Model.replaceById(id, data, options, cb);
+};
+
+DataAccessObject.replaceById = function(id, data, options, cb) {
+  var connectionPromise = stillConnecting(this.getDataSource(), this, arguments);
+  if (connectionPromise) {
+    return connectionPromise;
+  }
+  var pkName = idName(this);
+  if (!data[pkName]) data[pkName] = id;
+
+  var Model = this;
+  var inst = new Model(data);
+  var enforced = {};
+  this.applyProperties(enforced, inst);
+  inst.setAttributes(enforced);
+  Model = this.lookupModel(data); // data-specific
+  if (Model !== inst.constructor) inst = new Model(data);
+  var strict = inst.__strict;
+
+  assert((options || cb), 'The cb must be passed');
+
+  if (cb === undefined) {
+    if (typeof options === 'function') {
+      cb = options;
+      options = {};
+    }
+  }
+
+  cb = cb || utils.createPromiseCallback();
+  options = options || {};
+
+  assert((typeof data === 'object') && (data !== null),
+          'The data argument must be an object');
+  assert(typeof options === 'object', 'The options argument must be an object');
+  assert(typeof cb === 'function', 'The cb argument must be a function');
```

Comment:
Please move the code massaging input arguments and checking their validity at the top of the method, just after `stillConnecting` call.